cmake/kokkos_tribits.cmake

```diff
@@ -160,6 +160,12 @@ FUNCTION(KOKKOS_ADD_EXECUTABLE_AND_TEST ROOT_NAME)
             )
         ENDIF()
     ENDIF()
+    # We noticed problems with -fvisibility=hidden for inline static variables
+    # if Kokkos was built as shared library.
+    IF(BUILD_SHARED_LIBS)
+      SET_PROPERTY(TARGET ${PACKAGE_NAME}_${ROOT_NAME} PROPERTY VISIBILITY_INLINES_HIDDEN ON)
+      SET_PROPERTY(TARGET ${PACKAGE_NAME}_${ROOT_NAME} PROPERTY CXX_VISIBILITY_PRESET hidden)
```

Comment:
Why did you chose set target properties rather than set the CMAKE_ prefixed variables globally?

Reply:
I preferred the per-target property to better control which targets should use it. It needs precisely to be the test targets and none of the implementation files. With CMake variables this feels harder to control.

Reply:
Yeah I think we should avoid setting CMAKE_* variables if possible

Reply:
Why are you saying that.  I am talking about passing it in a few CI builds as argument, not setting it in our cmake files.